test/common/upstream/health_checker_impl_test_utils.h

```diff
@@ -97,30 +115,40 @@ class HttpHealthCheckerImplTest : public testing::Test, public HealthCheckerTest
 
   void setupServiceValidationWithoutUserAgent();
 
-  void expectSessionCreate(const HostWithHealthCheckMap& health_check_map);
-
-  void expectClientCreate(size_t index, const HostWithHealthCheckMap& health_check_map);
-
-  void expectStreamCreate(size_t index);
-
   void respond(size_t index, const std::string& code, bool conn_close, bool proxy_close = false,
                bool body = false, bool trailers = false,
                const absl::optional<std::string>& service_cluster = absl::optional<std::string>(),
                bool degraded = false);
 
-  void expectSessionCreate();
-  void expectClientCreate(size_t index);
-
   void expectSuccessStartFailedFailFirst(
       const absl::optional<std::string>& health_checked_cluster = absl::optional<std::string>());
 
   MOCK_METHOD(void, onHostStatus, (HostSharedPtr host, HealthTransition changed_state));
+};
 
-  std::vector<TestSessionPtr> test_sessions_;
-  std::shared_ptr<TestHttpHealthCheckerImpl> health_checker_;
-  std::list<uint32_t> connection_index_{};
-  std::list<uint32_t> codec_index_{};
-  const HostWithHealthCheckMap health_checker_map_{};
+class TcpHealthCheckerImplTestBase : public HealthCheckerTestBase {
+public:
+  void expectClientCreate();
+
+  std::shared_ptr<TcpHealthCheckerImpl> health_checker_;
+  Network::MockClientConnection* connection_{};
+  Network::ReadFilterSharedPtr read_filter_;
+};
+
+class TcpHealthCheckerImplTest : public testing::Test, public TcpHealthCheckerImplTestBase {
```

Comment:
if this class isn't planning on being re-used for the fuzz test, keep in the unit test (the same for HttpHealthCheckerImplTest, but you may want to move that in a separate clean-up PR)